homeassistant/data_entry_flow.py

```diff
@@ -348,6 +361,29 @@ def async_external_step_done(self, *, next_step_id: str) -> Dict[str, Any]:
             "step_id": next_step_id,
         }
 
+    @callback
+    def async_info_step(
```

Comment:
Let's call this `async_show_progress`.

I think that it will be good if we also include a `progress_action`, a placeholder for a translation to describe what is currently happening. 

For example, if our description is "Installing the OZW add-on". The current step could be "Downloading", "Installing", "Configuring".

Reply:
How will that word be translated? Usually the placeholders are not translated, right?

Reply:
Sorry, I should not have used the word placeholder but "translation key" :) Will work just like we do field names. So we can add it to strings.json like we do abort/error.

```json
"progress": {
  "install": "Installing"
}
```

Reply:
Right. Nice!

Reply:
I made `progress_action` a string similar to abort reason. I'm thinking each show progress will only be caused by a single action.